frame/contracts/src/lib.rs

```diff
@@ -942,11 +946,14 @@ where
 	}
 
 	/// Query storage of a specified contract under a specified key.
-	pub fn get_storage(address: T::AccountId, key: [u8; 32]) -> GetStorageResult {
+	pub fn get_storage(address: T::AccountId, key: Vec<u8>) -> GetStorageResult {
 		let contract_info =
 			ContractInfoOf::<T>::get(&address).ok_or(ContractAccessError::DoesntExist)?;
 
-		let maybe_value = Storage::<T>::read(&contract_info.trie_id, &key);
+		let maybe_value = Storage::<T>::read(
+			&contract_info.trie_id,
+			&VarSizedKey::<T>::try_from(key).map_err(|_| ContractAccessError::DoesntExist)?,
```

Comment:
This error doesn't exactly fit what is happening here. You might want to add another one. Even better: Already use a `VarSizedKey` as an argument here to push the error further out.